src/views/CreateEventView.vue

```diff
@@ -88,17 +106,43 @@ export default {
         description: String,
         whatToBring: String,
       */
+      imageUploaded: 0,
       event: {},
       error: '',
     };
   },
   methods: {
+    convertImage(file) {
+      return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => resolve(reader.result);
+        reader.onerror = error => reject(error);
+      });
+    },
+    updateEventImage(event) {
+      const { files } = event.target;
+      if (files.length > 0) {
+        this.imageUploaded = 1;
+        this.convertImage(files[0]).then((result) => {
+          this.imageUploaded = 2;
+          this.event = {
+            ...this.event,
+            thumbnail: result.toString(),
+          };
+        }).catch(() => {
+          this.imageUploaded = 3;
+        });
+      }
+    },
     onSubmit() {
       this.$validator.validateAll().then(async (result) => {
-        if (result) {
+        if (result && this.imageUploaded) {
           try {
-            await api.createEvent(this.event);
-            this.event = {};
+            const resp = await api.createEvent(this.event);
+            if (resp.status && resp.status === 200) {
```

Comment:
We should have optional chaining implemented in this repo? I think?
`resp.?status == 200` is equivalent

Reply:
Ah, good to know